ci-operator/step-registry/aws-load-balancer/pre-install-rosa/aws-load-balancer-pre-install-rosa-commands.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+AWSCRED="${CLUSTER_PROFILE_DIR}/.awscred"
+ROSA_TOKEN=$(cat "${CLUSTER_PROFILE_DIR}/ocm-token")
+# Use mounted registry-pull-credentials secret.
+# $CLUSTER_PROFILE_DIR/pull-secret cannot be used here.
+# It doesn't have the auths for the CI build registries (e.g. registry.build01.ci.openshift.org).
+CI_REGISTRY_PULL_SECRET="/var/run/albo/registry/.dockerconfigjson"
+REGION="${LEASED_RESOURCE}"
+CLUSTER_ID=$(cat "${SHARED_DIR}/cluster-id")
+# ALBO roles have to be cleaned up, otherwise ccoctl won't create secret manifests.
+# IAM roles with the prefix below are cleaned up by rosa-sts-account-roles-delete ref.
+ACCOUNT_ROLES_PREFIX=$(cat "${SHARED_DIR}/account-roles-prefix")
+CR_DIR="${SHARED_DIR}"
+MANIFEST_DIR="/tmp/albo-manifests"
+E2E_INPUT_DIR="${SHARED_DIR}"
+E2E_WAFV2_WEB_ACL_NAME="echoserver-acl"
+E2E_WAF_WEB_ACL_NAME="echoserverclassicacl"
+
+echo "=> configuring aws"
+if [ -f "${AWSCRED}" ]; then
+  export AWS_SHARED_CREDENTIALS_FILE="${AWSCRED}"
+  export AWS_DEFAULT_REGION="${REGION}"
+else
+  echo "Did not find compatible cloud provider cluster_profile"; exit 1
+fi
+
+echo "=> logging into rosa"
+if [ ! -z "${ROSA_TOKEN}" ]; then
+  echo "Logging into staging with offline token using rosa cli $(rosa version)"
+  rosa login --env "staging" --token "${ROSA_TOKEN}" || { echo "Login failed"; exit 1; }
+else
+  echo "Cannot login! You need to specify the offline token ROSA_TOKEN!"; exit 1
+fi
+
+echo "=> getting identity provider name"
+rosa describe cluster --cluster="${CLUSTER_ID}"
+AWS_ACCOUNT_ID=$(rosa describe cluster --cluster="${CLUSTER_ID}" | grep 'AWS Account:' | tr -d ' ' | cut -d: -f2)
+OIDC_IDP_NAME=$(rosa describe cluster --cluster="${CLUSTER_ID}" | grep 'OIDC Endpoint URL:' | tr -d ' ' | cut -d: -f2- | cut -d/ -f3-)
+IDP_ARN="arn:aws:iam::${AWS_ACCOUNT_ID}:oidc-provider/${OIDC_IDP_NAME}"
+
+echo "=> extracting ccoctl binary from cco image"
+CCO_IMAGE=$(oc adm release info --registry-config=${CI_REGISTRY_PULL_SECRET} --image-for='cloud-credential-operator' "${RELEASE_IMAGE_LATEST}")
```

Comment:
Can the pull secret have problematic characters?
```suggestion
CCO_IMAGE=$(oc adm release info --registry-config="${CI_REGISTRY_PULL_SECRET}" --image-for='cloud-credential-operator' "${RELEASE_IMAGE_LATEST}")
```

Reply:
`CI_REGISTRY_PULL_SECRET` contains the dockerconfig path, not dockerconfig contents.

Reply:
Done.